api/v1alpha3/types.go

```diff
@@ -381,6 +388,10 @@ type SubnetSpec struct {
 	// RouteTable defines the route table that should be attached to this subnet.
 	// +optional
 	RouteTable RouteTable `json:"routeTable,omitempty"`
+
+	// NatGateway associated with this subnet.
+	// +optional
+	NatGateway NatGateway `json:"natGateway,omitempty"`
```

Comment:
where is this used? you also need to update the controller to consume this. See how route table is created/set on the subnet.

Reply:
This is not used. I figured it would be better to first introduce the field and then make the change in the relevant controllers. Should I submit all the changes in this PR instead?

Reply:
I personally would prefer we don't merge unused API fields. Feel free to separate the controller changes in a separate commit in the same PR.